src/core/border_router/routing_manager.cpp

```diff
@@ -628,16 +629,23 @@ void RoutingManager::SendRouterAdvertisement(RouterAdvTxMode aRaTxMode)
         SuccessOrExit(error = mRioAdvertiser.AppendRios(raMsg));
     }
 
+    if (Get<InfraIf>().GetLinkLayerAddress(linkAddr) == kErrorNone)
+    {
+        SuccessOrExit(error = raMsg.AppendLinkLayerOption(linkAddr, Option::Type::kSourceLinkLayerAddr));
```

Comment:
Suggest removing the `Type::` when using the `enum` values. Since it's a normal `enum` definition (and not an `enum class`), we don't need the `Type::`. Most toolchains will handle and ignore it, but some older ones may give errors or warnings, so we generally avoid using this model to refer to the enum values. 

```suggestion
        SuccessOrExit(error = raMsg.AppendLinkLayerOption(linkAddr, Option::kSourceLinkLayerAddr));
```